### PR TITLE
core(is-crawlable): only warn if some bots are blocked

### PR DESCRIPTION
**Pull request by connorjclark**
ref #5329 (can't close until we update docs)

This audit will no longer fail if some known crawler is explicitly allowed while others are blocked. Instead, it will emit a warning say which user agents have been blocked. Details items are only present when the audit is failing, and only show blocking directives wrt a generic UA.

here's a really big list of crawler UAs ... https://github.com/monperrus/crawler-user-agents/blob/master/crawler-user-agents.json . What all should we explicitly look for?

### CODE REVIEW

File: core/test/audits/seo/is-crawlable-test.js [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 import assert from 'assert/strict';
 
+import {expect} from 'expect';
```
Comment (adamraine) [inline core/test/audits/seo/is-crawlable-test.js:9]: This is already a global right?

File: core/audits/seo/is-crawlable.js [hunk 2]
```diff
@@ -85,64 +95,126 @@ class IsCrawlable extends Audit {
...
-          source: {
-            type: /** @type {const} */ ('source-location'),
-            url: robotsFileUrl.href,
-            urlProvider: /** @type {const} */ ('network'),
-            line: line - 1,
-            column: 0,
-          },
-        });
-      }
+    const score = blockedUserAgents.length !== BOT_USER_AGENTS.size ? 1 : 0;
```
Comment (adamraine) [inline core/audits/seo/is-crawlable.js:200]: nit
```suggestion
    const score = blockedUserAgents.length === BOT_USER_AGENTS.size ? 0 : 1;
```

**Review by adamraine — APPROVED** [review]

File: core/audits/seo/is-crawlable.js [hunk 3]
```diff
@@ -48,7 +56,7 @@ function isUnavailable(directive) {
 
 /**
  * Returns true if any of provided directives blocks page from being indexed
- * @param {string} directives
+ * @param {string} directives assumes no user-agent prefix
```
[the remark this answers is not in the document]
Comment (connorjclark) [reply core/audits/seo/is-crawlable.js:59]: This is proper jsdoc syntax. IDE makes it look like this on hover:
image


and not hovered color should be different

image

File: core/test/audits/seo/is-crawlable-test.js (hunk 1 above)
Comment (connorjclark) [reply core/test/audits/seo/is-crawlable-test.js]: yeah. i removed from other places too just now

**Pull request merged**